packages/protocol/contracts/tokenvault/ERC20Vault.sol

```diff
@@ -159,7 +165,10 @@ contract ERC20Vault is BaseVault {
         nonReentrant
         returns (address btokenOld_)
     {
-        if (_btokenNew == address(0) || bridgedToCanonical[_btokenNew].addr != address(0)) {
+        if (
+            _btokenNew == address(0) || bridgedToCanonical[_btokenNew].addr != address(0)
+                || !_btokenNew.supportsInterface(type(IBridgedERC20).interfaceId)
```

Comment:
`_btokenNew.supportsInterface(type(IBridgedERC20).interfaceId)` -> How the USDC contract (which we plan to deploy right after genesis block) will pass this check ?
This script will fail, if not mistaken:
https://github.com/taikoxyz/taiko-mono/pull/16812/files

Reply:
Good finding. Removed this and IERC165 related functions.